tests/scripts/all.sh

```diff
@@ -1608,6 +1608,15 @@ component_test_memsan_constant_flow () {
     make test
 }
 
+skip_non_constant_flow_components () {
```

Comment:
This is not actually having any effect. Setting `SKIP_TEST_SUITES` in the environment only works with make, not with cmake.

`tests/CMakeLists.txt` does have code to support `SKIP_TEST_SUITES`. I added the feature in https://github.com/Mbed-TLS/mbedtls/pull/2293 and I'm pretty sure it worked at the time, but it seems to have bitrotted. I use the feature often with make, but never with cmake.

Reply:
I put up #6676 which gets `SKIP_TEST_SUITES` from the environment if it's not defined directly with `-D`. That should allow this PR to work.